### ISSUE TITLE
Don't let ContentsHandler's internal 404 bubble up.

### ISSUE THREAD

**Issue by Carreau** [jupyter-server/jupyter_server#1328]
See https://github.com/jupyterlab/jupyterlab/issues/15153

ContentsHandler's get requests (and future HEAD requests https://github.com/jupyter-server/jupyter_server/issues/1327) should likely catch internal 404s and return a clean 404 with 'set_status()' and `finish()` this will avoid tornado to print tracebacks on non-existing files, which are often requested on purpose, like in `SaveAs`.

**Comment by Zsailer** [comment]
Yeah, I agree here, @Carreau. Thank you for raising here. I've labeled the issue with "good first issue" since this should be fairly strait forward.

**Comment by bloomsa** [comment]
can this be assigned to me? I'll give it a shot

**Comment by bloomsa** [comment]
FYI I think this can be closed now that #1357 is merged. let me know if I'm wrong
